satellite/discovery/service.go

```diff
@@ -131,6 +134,11 @@ func (discovery *Discovery) refresh(ctx context.Context) (err error) {
 				if _, err = discovery.cache.UpdateNodeInfo(ctx, node.Id, info); err != nil {
 					discovery.log.Warn("could not update node info", zap.Stringer("ID", node.GetAddress()))
 				}
+
+				// update/add the certdb with nodeID's public key
+				if err = discovery.certdb.SavePublicKey(ctx, node.Id, ident); err != nil {
```

Comment:
**`Rational for calling a SavePublicKey()`**
At the time of implementing this ticket the only place where the satellite can extract Storagenode's peer identity is in Kademlia. When Kad implementation changes, only this part changes and rest of the frame work doesn't need any major changes.